core/src/replay-parse/finalization/finalization.service.ts

```diff
@@ -56,19 +58,26 @@ export class FinalizationService {
         if (scrimResponse.status === ResponseStatus.ERROR) throw scrimResponse.error;
         const scrimObject = scrimResponse.data;
 
-        await Promise.all([
+        return Promise.all([
             this.mledbScrimService.saveScrim(submission, submissionId, runner, scrimObject as Scrim),
             this.saveToSprocket(submission, runner, scrimObject as Scrim),
         ])
-            .then(async () => runner.commitTransaction())
+            .then(async ([mledbScrimId, sprocketMatchParentId]) => {
+                await runner.commitTransaction();
+
+                return {
+                    id: sprocketMatchParentId,
+                    legacyId: mledbScrimId,
+                };
+            })
             .catch(async e => {
                 await runner.rollbackTransaction();
                 this.logger.error(e);
                 throw e;
             });
```

Comment:
I know this is how it was previously, but can we refactor this to use async/await because we are in an async function?

Now that the return function does more than simply commit the transaction; it makes more sense to lay it out that way.